### PR TITLE
OCM-2342: Adding IDP examples and docs phase 1

### PR DESCRIPTION
**Pull request by sagidayan**
- Google
- GitHub
- GitLab
- htpasswd

### CODE REVIEW

**Review by tshefi — COMMENTED** [review]
Noticed two things, see inline.

File: examples/create_identity_provider/github/main.tf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  }
+}
+
+provider "ocm" {
+}
+
+resource "ocm_identity_provider" "github_idp" {
+  cluster = "my-cluster"
+  name    = "GitHub"
```
[the remark this answers is not in the document]
Comment (sagidayan) [reply examples/create_identity_provider/github/main.tf:35]: Will update the read me then. To have a consistent naming format.

File: examples/create_identity_provider/github/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+4. Optional: Enter an application description.
+    
+5. Enter the authorization callback URL, where the end of the URL contains the identity provider `name`:
+    
+    `https://oauth-openshift.apps.<cluster-name>.<cluster-domain>/oauth2callback/<idp-provider-name>`
+    
+    For example:
+    `https://oauth-openshift.apps.openshift-cluster.example.com/oauth2callback/Github`
+	
+	> **Note**: `<idp-provider-name>` is case-sensitive.
```
Comment (nirarg) [inline examples/create_identity_provider/github/README.md:39]: Add here link to the place in the example which must be identical
In the example you used GitHUb

File: examples/create_identity_provider/gitlab/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+## Prerequisites
+
+### An OpenShift cluster
+
+This example assumes you have created a cluster via OCM, It can be via this terraform provider or by a different client.
+We will need the cluster name and some credentials.
+
+### Setting up your application in GitLab
+
+You will need a client ID/Secret of a [registered GitLab OAuth application](https://docs.gitlab.com/ce/api/oauth2.html). The application must be configured with a callback URL of `https://oauth-openshift.apps.<cluster-name>.<cluster-domain>/oauth2callback/<idp-provider-name>`
```
Comment (nirarg) [inline examples/create_identity_provider/gitlab/README.md:17]: In Github you added here the actual name and here you added `<idp-provider-name>`
Please be consistent

File: examples/create_identity_provider/google/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+### An OpenShift cluster
+
+This example assumes you have created a cluster via OCM, It can be via this terraform provider or by a different client.
+We will need the cluster name and some credentials.
+
+### Setting up your application in Google
+
+You will need a client ID/Secret of a registered Google project.
+The project must be configured with a redirect URI of `https://oauth-openshift.apps.<cluster-name>.<cluster-domain>/oauth2callback/<idp-provider-name>`.
+> **Note**: `<idp-provider-name>` is case-sensitive.
```
Comment (nirarg) [inline examples/create_identity_provider/google/README.md:19]: Where do you set this?
Can you provide link/instructions?

File: examples/create_identity_provider/htpasswd/main.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+provider "ocm" {
+}
+
+resource "ocm_identity_provider" "htpasswd_idp" {
+  cluster = "my-cluster"
+  name    = "htpasswd"
+  htpasswd = {
+    username = var.htpasswd_username
+    password = var.htpasswd_password
```
[the remark this answers is not in the document]
Comment (sagidayan) [reply examples/create_identity_provider/htpasswd/main.tf:38]: We do not validate it in the provider at the moment.

File: examples/create_identity_provider/gitlab/README.md (hunk 3 above)
Comment (sagidayan) [reply examples/create_identity_provider/gitlab/README.md]: Ill add the example here as well

File: examples/create_identity_provider/google/README.md (hunk 4 above)
Comment (sagidayan) [reply examples/create_identity_provider/google/README.md]: Added a link

**Pull request merged**